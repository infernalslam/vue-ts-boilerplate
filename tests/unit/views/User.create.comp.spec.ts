import { shallowMount } from '@vue/test-utils';
import UserCreate from '@/views/User/Create.vue';

describe('User : Create.vue', () => {
  it('init component properly', () => {
    // const onBtnClick = sinon.stub();
    const onBtnClick = jest.fn().mockReturnValue('');
    const wrapper = shallowMount(UserCreate, {
      methods: {
        onBtnClick,
      },
    });

    wrapper.find('#btn').trigger('click');
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    expect(onBtnClick).toHaveBeenCalled();
  });
});
