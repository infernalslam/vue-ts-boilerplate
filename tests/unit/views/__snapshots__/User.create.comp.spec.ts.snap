// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User : Create.vue init component properly 1`] = `
<div
  class="user-create"
>
  <h3>
    User Create
  </h3>
   
  <button
    id="btn"
  >
    Click
  </button>
   
  <hr
    class="line"
  />
   
  <h3>
    Created User
  </h3>
   
  <h3>
    
  </h3>
</div>
`;
